cpp/src/decisiontree/decisiontree.cu

```diff
@@ -172,8 +184,20 @@ void decisionTreeClassifierFit(const raft::handle_t &handle,
                                uint64_t seed) {
   std::shared_ptr<DecisionTreeClassifier<double>> dt_classifier =
     std::make_shared<DecisionTreeClassifier<double>>();
+  MLCommon::device_buffer<double> *global_quantiles = nullptr;
```

Comment:
minor nitpick: please use `std::unique_ptr` instead to automatically take care of memory lifetimes. Then you wouldn't also need to do the explicit `release` and `delete` below.